js/visearch.js

```diff
@@ -132,13 +133,24 @@ const RESULT_LOAD = 'result_load';
       }
     }
 
+    /**
+     * Save the query Id of the last request to local storage
+     * @param {*} resp response from ProductSearch API
+     */
+    function saveQueryId(resp) {
+      queryId = resp.reqid;
```

Comment:
can we only store the success queryId?
we can make use of the `status` field if === 'OK'

Reply:
can refer to the send result load check
```function sendResultLoadEvent(productId, args) {
      if (args.status !== 'OK' || args.result.length === 0) {
        return;
      }```